win32ss/user/user32/windows/message.c

```diff
@@ -2900,12 +2900,14 @@ WINAPI
 RealGetQueueStatus(UINT flags)
 {
    #define QS_TEMPALLINPUT 255 // ATM, do not support QS_RAWINPUT
+   DWORD ret;
    if (flags & ~(QS_SMRESULT|QS_ALLPOSTMESSAGE|QS_TEMPALLINPUT))
```

Comment:
@julcar, I compared the wine code that you linked and it appears that if you only change this statement you can leave all of the other code in place. The new line should be as follows:
```
   if (flags & ~(QS_SMRESULT|QS_ALLPOSTMESSAGE|QS_ALLINPUT))
```
It appears that your real effect was not ever returning zero (0) and in our case, this was a very expensive thing.
Thanks. Doug Lyons

Reply:
Right, here seems that the magic fix here is avoiding the function to return 0 and that may explain why just casting the return to DWORD was not enough to fix the bug.